docs/docs/adrs/adr-018-permissionless-ics.md

```diff
@@ -0,0 +1,270 @@
+  ibc.core.client.v1.Height initial_height;
+
+  ...  (all fields contained in ConsumerAdditionProposal except top_N)
+  
+  uint32 validators_power_cap;
+  uint32 validator_set_cap;
+  repeated string allowlist;
+  repeated string denylist;
+}
+```
+
+`MsgLaunchConsumerChain` contains everything that is contained in [`ConsumerAdditionProposal`](https://github.com/cosmos/interchain-security/blob/v4.3.0/proto/interchain_security/ccv/provider/v1/provider.proto#L29)
+except the `top_N` field because we can only launch Opt In consumer chains with this message (Top N chains still need to go
+through a governance proposal). Note, that an alternative format for this message could have been to include a single `ConsumerAdditionProposal` field in it,
+but this confuses the semantics of `MsgLaunchConsumerChain` and that of a governance proposal. The response of this message
+(i.e., `MsgLaunchConsumerChainResponse`) contains a single `string` that is the `consumerID`. The `consumerID`s corresponds
+to `len(chainID) | chainID | counter` and this way we can have multiple different `consumerID`s that correspond to a chain
+with the same `chainID`. By doing this, we tackle the main issue behind permissionless consumer chains that is chain-id squatting.
+Note that `consumerID` could just be a `uint64` but we choose to include `chainID` in it as well so that it is easier to see
+what the consumer chain just by looking at the `consumerID`. This means that the `chainID` of a chain cannot be changed
+after launching it (because the `chainID` is part of the `consumerID` key).
+
+We intend to set a fixed cost of a `MsgLaunchConsumerChain` to avoid getting spammed with bogus consumer launch chains.
+
+To execute a `MsgLaunchConsumerChain`, we first create a `ConsumerAdditionProposal` under the hoods, with the `top_N` set to 0, and call
+[`HandleConsumerAdditionProposal`](https://github.com/cosmos/interchain-security/blob/v4.3.0/x/ccv/provider/keeper/proposal.go#L30)
+simulating a proposal that was already voted on and accepted. Note that we need to migrate `ConsumerAdditionProposal`s to use
+the `consumerID` instead using [both](https://github.com/cosmos/interchain-security/blob/v4.3.0/x/ccv/provider/keeper/proposal.go#L382)
+`chainID` and the `spawnTime` as keys. The [usual validity conditions](https://github.com/cosmos/interchain-security/blob/v4.3.0/x/ccv/provider/types/proposal.go#L114)
+hold for the fields of `MsgLaunchConsumerChain`. Note however, that we intend to have a `spawnTime` upper limit as well.
+For example, if you launch a consumer chain in Permissionless ICS, the `spawnTime` should not be more
+than two months ahead in the future, to avoid having consumer chains lingering for too long before they get added. To do this,
+we introduce a `maxSpawnTime` limit in [`ConsumerAdditionProposal`](https://github.com/cosmos/interchain-security/blob/v4.3.0/proto/interchain_security/ccv/provider/v1/provider.proto#L29).
```

Comment:
How is the `maxSpawnTime` set? For proposals, they can choose it themselves, and for launches via TX, it is set by current time + some parameter?

Reply:
Removed the maximum `spawnTime`. We iterate chains to be launched by ascending order based on their `spawnTime` so we do not have to iterate through all of them in any case.